queries.js

```diff
@@ -0,0 +1,171 @@
+const dayjs = require('dayjs');
+
+module.exports = (self, query) => {
+  return {
+    builders: {
+      future: {
+        def: null,
+        async finalize() {
+          const future = query.get('future');
+          if (future === null || query.get('year') || query.get('month')) {
```

Comment:
Presence of year/month/day should not bypass `future`. The general public should not be able to sneak their way into unreleased territory at all. Only future === null does that.

Reply:
Done